RZAssemblage/RZArrayAssemblage.m

```diff
@@ -0,0 +1,170 @@
+//
+//  RZArrayAssemblage.m
+//  RZAssemblage
+//
+//  Created by Brian King on 3/21/15.
+//  Copyright (c) 2015 Raizlabs. All rights reserved.
+//
+
+#import "RZArrayAssemblage.h"
+#import "RZAssemblage+Private.h"
+
+NSString *const RZAssemblageUpdateKey = @"RZAssemblageUpdateKey";
+static char RZAssemblageUpdateContext;
+
+@implementation RZArrayAssemblage
+
++ (BOOL)shouldObserveContents
+{
+    return YES;
+}
+
+- (instancetype)initWithArray:(NSArray *)array representingObject:(id)representingObject;
+{
+    self = [super init];
+    if ( self ) {
+        self.representedObject = representingObject;
+        _childrenStorage = [array isKindOfClass:[NSMutableArray class]] ? array : [array mutableCopy];
+        if ( self.class.shouldObserveContents ) {
+            [[_childrenStorage copy] enumerateObjectsUsingBlock:^(id object, NSUInteger idx, BOOL *stop) {
+                [self addMonitorsForObject:object];
```

Comment:
I was hoping to avoid making a mutable subclass here, and I think it's more crufty than it's worth.

That copy isn't needed, should revert to a for loop.